osu.Framework/Platform/GameHost.cs

```
@@ -532,12 +533,42 @@ protected virtual void DrawFrame()
         /// </summary>
         protected virtual void Swap()
         {
-            Window.SwapBuffers();
+            if(currentWorkarouns.HasFlagFast(PlatformWorkaround.WindowsInvalidateRect)) // Should only be set on Windows platforms
+            {
+                // InvalidateRect is absolutely necessary for some driver workarounds,
+                //  otherwise the latency will be higher, and dwm might crash sometimes
+                if (this.Window is SDL2DesktopWindow sdlWindow)
+                    Windows.Native.Window.InvalidateRect(sdlWindow.WindowHandle, IntPtr.Zero, false);
+            }
 
-            if (Window.VerticalSync)
-                // without glFinish, vsync is basically unplayable due to the extra latency introduced.
-                // we will likely want to give the user control over this in the future as an advanced setting.
+            if(currentWorkarouns.HasFlagFast(PlatformWorkaround.FinishBeforeSwap))
+            {
+                //TODO: This glFinish is only needed because we're not able to call wglSwapLayerBuffers directly, as to not break any possible future SDL code changes
                 GL.Finish();
+            }
+
+            Window.SwapBuffers();
+
+            if(currentWorkarouns.HasAnyFlagFast(PlatformWorkaround.FinishAfterSwapVSync | PlatformWorkaround.FinishAfterSwapNoVSync))
+            {
+                if(Window.VerticalSync)
```

Please check code style.

---

This is embarrassing. It seems like the style checker is silently failing to run on my end.